server/src/main/java/org/opensearch/index/codec/composite/composite912/Composite912DocValuesWriter.java

```diff
@@ -148,6 +144,19 @@ public Composite912DocValuesWriter(DocValuesConsumer delegate, SegmentWriteState
         segmentHasCompositeFields = Collections.disjoint(segmentFieldSet, compositeFieldSet) == false;
     }
 
+    private void addStarTreeSupportedFieldsFromSegment() {
+        // TODO : add integ test for this
+        for (FieldInfo fi : this.state.fieldInfos) {
```

Comment:
might be missing something but we are doing the same thing in all the if else conditions right?